content/en/docs/concepts/architecture/mixed-version-proxy.md

```diff
@@ -82,29 +106,16 @@ loads a special filter that does the following:
 ### How it works under the hood
 
 When an API Server receives a resource request, it first checks which API servers can
-serve the requested resource. This check happens using the internal
-[`StorageVersion` API](/docs/reference/generated/kubernetes-api/v{{< skew currentVersion >}}/#storageversioncondition-v1alpha1-internal-apiserver-k8s-io).
-
-* If the resource is known to the API server that received the request
-  (for example, `GET /api/v1/pods/some-pod`), the request is handled locally.
-
-* If there is no internal `StorageVersion` object found for the requested resource
-  (for example, `GET /my-api/v1/my-resource`) and the configured APIService specifies proxying
-  to an extension API server, that proxying happens following the usual
-  [flow](/docs/tasks/extend-kubernetes/configure-aggregation-layer/) for extension APIs.
-
-* If a valid internal `StorageVersion` object is found for the requested resource
-  (for example, `GET /batch/v1/jobs`) and the API server trying to handle the request
-  (the _handling API server_) has the `batch` API disabled, then the _handling API server_
-  fetches the peer API servers that do serve the relevant API group / version / resource
-  (`api/v1/batch` in this case) using the information in the fetched `StorageVersion` object.
-  The _handling API server_ then proxies the request to one of the matching peer kube-apiservers
-  that are aware of the requested resource.
-
-  * If there is no peer known for that API group / version / resource, the handling API server
-    passes the request to its own handler chain which should eventually return a 404 ("Not Found") response.
-
-  * If the handling API server has identified and selected a peer API server, but that peer fails
-    to respond (for reasons such as network connectivity issues, or a data race between the request
-    being received and a controller registering the peer's info into the control plane), then the handling
-    API server responds with a 503 ("Service Unavailable") error.
+serve the requested resource. This check happens using the non peer-aggregated discovery document.
+
+* If the resource is listed in the non peer-aggregated discovery document retrieved from the API server that received the request(for example, `GET /api/v1/pods/some-pod`), the request is handled locally.
+
+* If the resource in a request (for example, `GET /batch/v1/jobs`) is not found in the non peer-aggregated discovery document retrieved from the API server trying to handle the request (the _handling API server_), likely because it has the `batch` API disabled, then the _handling API server_ fetches the peer API servers that do serve the relevant API group / version / resource (`batch/v1` in this case) by checking the non peer-aggregated discovery documents from all peer API servers. The _handling API server_ then proxies the request to one of the matching peer kube-apiservers that are aware of the requested resource.
```

Comment:
nit: I know this is pre-existing, but giving the example that an apiserver intentionally disables a GA group-version `batch` feels weird. Can we be consistent with the upgrade scenario and say something like `jobs` is a new API added and the old apiserver does not support it?

Reply:
Updated to use an example for `/apis/resource.k8s.io/v1beta1/resourceclaims` and clarified that the resource may not be supported on an older apiserver because it may have been introduced in a later version.